platform/default/include/mbgl/util/default_thread_pool.hpp

```diff
@@ -6,17 +6,40 @@
 #include <mutex>
 #include <queue>
 #include <thread>
+#include <functional>
 
 namespace mbgl {
 
+struct ThreadLifecycle {
+    using ThreadData = void*;
+    ThreadLifecycle() = default;
+    ThreadLifecycle(std::function<ThreadData()> onThreadCreated_,
+                    std::function<void(ThreadData)> onThreadDestroyed_) :
+            onThreadCreatedFn(onThreadCreated_),
+            onThreadDestroyedFn(onThreadDestroyed_) {}
+
+    ThreadData onThreadCreated() const {
```

Comment:
I would keep `ThreadData` private to `ThreadLifecycle`, unless some other client needs it.